### PR TITLE
Spec integration test

### PR DESCRIPTION
**Pull request by serengia**
Kindly review my request to merge `spec_integration_test` branch to `development` branch.
- The branch contains the implementation for solving n+1 problem on queries
- Implementation of integration tests for the posts and users resources 
Before
Screenshot (23)
After
Screenshot (25)

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,7 +1,8 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = @user.posts
+    # Eager loading comments and likes for the user's posts
+    @posts = @user.posts.includes(:comments, :likes)
```
Comment (danmainah) [inline app/controllers/posts_controller.rb:6]: - [ ] @Team great work fixing N +1 problem, however, you forgot to add screenshots of your app's logs before and after the fix in the pull request description, kindly consider implementing this, its part of the project requirement as shown below
image

File: spec/integration/user_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    it 'should render user information' do
+      expect(page).to have_content(@cristiano.name)
+      expect(page).to have_content(@john.name)
+
+      expect(page).to have_css("img[src*='https://picsum.photos/200/300']")
+      expect(page).to have_css("img[src*='default_user.jpg']")
+
+      expect(page).to have_content(@cristiano.posts_counter)
+      expect(page).to have_content(@john.posts_counter)
```
Comment (danmainah) [inline spec/integration/user_index_spec.rb]: - [ ] @Team great work adding tests for the user index page, however, the tests that assert you can see the username,  and profile picture are supposed to check that each user profile picture and username is visible, kindly consider using each loop in your tests as you did in posts show page to achieve the objective.
- [ ] @Team you also forgot to add a test that assets ` you can see the number of posts the user has written.`
image

File: spec/integration/post_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      @messi_post1.five_recent_comments.each do |comment|
+        expect(page).to have_content(comment.text)
+      end
+    end
+
+    it 'should redirect to post show page when clicking on post title' do
+      click_link @messi_post1.title.capitalize
+      expect(page).to have_current_path(user_post_path(@messi, @messi_post1))
+    end
+  end
```
Comment (danmainah) [inline spec/integration/post_index_spec.rb:59]: - [ ] @Team great work adding tests for the posts page, however, you forgot to add a test that assets a user `can see a section for pagination if there are more posts than fit on the view.`, its part of the project requirement as shown below
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @serengia , @DJ-MrJay , and @Elhambasir , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Added tests for all the four pages
✔️  Fixed the N + 1 problem.
✔️ Professional written README file with all sections required.

# Required Changes ♻️

- [ ] @Team Kindly ensure all tests are passing after implementing the requested changes.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- @serengia  Kindly consider adding the authors info in your readme author section, as a way to appriciate their contribution to this project.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi @serengia 👋, 

Great effort implementing all the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 🎯 
✔️ Descriptive PR
✔️ Added integration tests 📌 
✔️ All tests are green 🟢 
✔️ No linter errors

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**